src/main/java/org/opensearch/security/dlic/rest/api/MultiTenancyConfigApiAction.java

```diff
@@ -202,4 +221,30 @@ private void updateAndValidatesValues(final ConfigV7 config, final JsonNode json
         }
     }
 
+    private List<DashboardSignInOption> getNewDashboardSignInOptions(JsonNode newOptions) {
+        List<DashboardSignInOption> options = new ArrayList<>();
+        for (int i = 0; i < newOptions.size(); i++) {
+            try {
+                String option = newOptions.get(i).asText();
+                options.add(DashboardSignInOption.valueOf(option));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Invalid sign-in option. " + e.getMessage());
+            }
+        }
+        return options;
+    }
+
+    /**
+     * It enables/disables the authentication options available just for security dashboard sign-in options.
+     */
+    private void toggleAuthcDashboardSignInAbility(Authc authc, List<DashboardSignInOption> options, boolean enable) {
+        for (DashboardSignInOption option : options) {
+            for (String key : authc.getDomains().keySet()) {
+                if (key.contains(option.toString().toLowerCase())) {
+                    authc.getDomains().get(key).http_enabled = enable;
```

Comment:
Hi @cwperks, the assumption I made to change `http_enabled` is :
If an admin is using the Security Dashboard to enable or disable sign in options, and let's say the BASIC authentication is removed, other users still could get information from the API using an http requests.
In the image attached there are 3 request and you can see the following:

1. I made a request using admin and password and I got the information since I was authenticated correctly.
2. I disabled the BASIC authentication using Security Dashboard, tried to get the same information, and I got authentication failed since the option was disabled (isn't it the expected behavior?).
3. I removed the code you commented and as you can see even though the BASIC authentication is not among the available options I still got the information. isn't that bad? 

If I'm using Security Dashboard and I don't want the users authenticate using BASIC authentication, why should be the option still available?

![image](https://github.com/opensearch-project/security/assets/48450162/5460e434-af8c-4343-861e-956d01324d46)

Also, there is already an implementation in Security Dashboard where the option using to sign-in is validated before sending the request.

Reply:
> I removed the code you commented and as you can see even though the BASIC authentication is not among the available options I still got the information. isn't that bad?

Its not bad and its actually quite common. Some clusters may have SAML configured for sign in to dashboards, but also have basic auth where they can connect programmatically with an opensearch client. You're right that it is often the case that the authenticators enabled on the backend === the sign in options, but its not necessarily always the case and we should support that. 

This feature will exclusively control the sign in options that are displayed in OSD, but not enable or disable the authenticators on the backend. If a cluster admin wants to enable or disable an authenticator on the backend they would need to use the [security admin tool](https://opensearch.org/docs/latest/security/configuration/security-admin/) and connect with the admin certificate to apply changes.

Reply:
Oh, I understand now. @cwperks Thank you very much for the explanation.